server/src/main/java/org/opensearch/extensions/action/ExtensionMainAction.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.action.ActionType;
+
+/**
+ * The main proxy action for all extensions
```

Comment:
I withheld commenting on the "main" name figuring I'd get to it here.   My understanding of "main" across multiple programming languages is a primary executable starting point (the `main()` method.)

It's not clear to me that's the purpose here, it's not a starting point for anything.  So I might [quibble over the naming here](https://martinfowler.com/bliki/TwoHardThings.html).

Reply:
Haha. Sure what do you guys recommend? 
Few Options my brain can think of:
- `ExtensionProxyAction`
- `ExtensionAction`

Reply:
My vote is for `ExtensionProxyAction`